src/virt.c

```diff
@@ -761,6 +772,94 @@ static int lv_domain_info(virDomainPtr dom, struct lv_info *info) {
   return 0;
 }
 
+char *metadata_get_hostname(virDomainPtr dom) {
+  char *hostname = NULL;
+  const char *hostname_src = NULL;
+  size_t hostname_len = 0;
+  char *metadata_str = NULL;
+  const char *xpath_str = NULL;
+  const char *namespace = NULL;
+  xmlNodePtr xml_node = NULL;
+  xmlDocPtr xml_doc = NULL;
+  xmlXPathContextPtr xpath_ctx = NULL;
+  xmlXPathObjectPtr xpath_obj = NULL;
+
+  if (hm_xpath == NULL)
+    xpath_str = "/instance/name/text()";
+  else
+    xpath_str = hm_xpath;
+
+  if (hm_ns == NULL)
+  namespace = "http://openstack.org/xmlns/libvirt/nova/1.0";
+  else namespace = hm_ns;
```

Comment:
split into 2 lines

Reply:
clang-format don't like that, but I have surround the code with {} to make clang-format happy and have a well indented code.